core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -695,7 +696,7 @@ private[spark] class TaskSetManager(
   def canFetchMoreResults(size: Long): Boolean = sched.synchronized {
     totalResultSize += size
     calculatedTasks += 1
-    if (maxResultSize > 0 && totalResultSize > maxResultSize) {
+    if (!isShuffleMapTasks && maxResultSize > 0 && totalResultSize > maxResultSize) {
```

Comment:
This makes sense to me, as shuffle map task returns `MapStatus`. We won't keep the task results, we just update map output tracker, update accumulators, update metrics, and throw the task result away. So we don't need to sum the tasks result size and check it.

What do you think? @tgravescs @squito @holdenk  @jiangxb1987 @Ngone51

Reply:
According to the JIRA, does this affect only 3.0.0?

Reply:
It affects all the versions, but it's not a serious problem (users can set a very high max result size conf). I think we don't need to backport.

Reply:
Yeah, make sense to me, too.

Reply:
yes this seems reasonable.  It would be nice to update the comment on the function as well as put description in jira and in this PR to include more details about the shuffle map status and I assume you actually his this case in an application.

Reply:
LGTM too

Reply:
But seems we will save the MapStatus in MapOutputTracker. Don't we?

Reply:
Then we can count the MapStatus size only in the total task result size?

Reply:
> But seems we will save the MapStatus in MapOutputTracker. Don't we?

We do, but it's a global thing. Limiting the total size of one stage doesn't help too much. I believe the check here was to prevent too much data in `rdd.collect`.

Reply:
OK, make sense.